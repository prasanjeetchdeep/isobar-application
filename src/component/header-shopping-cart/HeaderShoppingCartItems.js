import React from 'react';
import "./HeaderShoppingCartItems.modules.scss";
import { ListGroup } from "react-bootstrap";

const HeaderShoppingCartItems = ({ book: { image, title, name, author }, index }) => {
    return (
        <ListGroup.Item key={`shoppingcart-${index}`} >
            <div className="flexbox header-item-container">
                <img src={image} alt={title} style={{ height: 30 }} />
                <div className="flexbox header-item-description">
                    <div className="flexbox header-item-name">{name}</div>
                    <div className="flexbox header-item-author">{author}</div>
                </div>
            </div>
        </ListGroup.Item>

    );
};

export default HeaderShoppingCartItems;

