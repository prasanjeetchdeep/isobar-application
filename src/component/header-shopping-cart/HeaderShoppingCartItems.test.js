import React from 'react';
import { render } from '@testing-library/react';
import HeaderShoppingCartItems from './HeaderShoppingCartItems';

const book= {
    id: 3,
    name: "CSS Animations",
    description: "Learn how to animate anything in CSS",
    author: "Alan Smith",
    publishDate: "04/12/2018",
    duration: "00:02:11",
    image:
      "https://cdn.pixabay.com/photo/2017/08/05/11/16/logo-2582747_960_720.png"
  };

it("renders the above book at index 1", () => {
    const { asFragment } = render(<HeaderShoppingCartItems book={book} index={1}/>);
    expect(asFragment()).toMatchSnapshot();
  });
