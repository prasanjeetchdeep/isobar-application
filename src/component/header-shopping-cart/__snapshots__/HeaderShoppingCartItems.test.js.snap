// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="list-group-item"
  >
    <div
      class="flexbox header-item-container"
    >
      <img
        src="https://cdn.pixabay.com/photo/2017/08/05/11/16/logo-2582747_960_720.png"
        style="height: 30px;"
      />
      <div
        class="flexbox header-item-description"
      >
        <div
          class="flexbox header-item-name"
        >
          CSS Animations
        </div>
        <div
          class="flexbox header-item-author"
        >
          Alan Smith
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders the above book at index 1 1`] = `
<DocumentFragment>
  <div
    class="list-group-item"
  >
    <div
      class="flexbox header-item-container"
    >
      <img
        src="https://cdn.pixabay.com/photo/2017/08/05/11/16/logo-2582747_960_720.png"
        style="height: 30px;"
      />
      <div
        class="flexbox header-item-description"
      >
        <div
          class="flexbox header-item-name"
        >
          CSS Animations
        </div>
        <div
          class="flexbox header-item-author"
        >
          Alan Smith
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
